tests/suite/snippets_filter_test.go

```diff
@@ -0,0 +1,337 @@
+			Expect(resourceManager.DeleteFromFiles(snippetsFilter, namespace)).To(Succeed())
+		})
+
+		Specify("snippetsFilters are accepted", func() {
+			snippetsFilterNames := []string{
+				"all-contexts",
+				"grpc-all-contexts",
+			}
+
+			for _, name := range snippetsFilterNames {
+				nsname := types.NamespacedName{Name: name, Namespace: namespace}
+
+				err := waitForSnippetsFilterToBeAccepted(
+					nsname,
+					metav1.ConditionTrue,
+					ngfAPI.SnippetsFilterConditionReasonAccepted,
+				)
+				Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("%s was not accepted", name))
+			}
+		})
+
+		Context("verify working traffic", func() {
+			It("should return a 200 response for HTTPRoute", func() {
+				port := 80
+				if portFwdPort != 0 {
+					port = portFwdPort
+				}
+				baseURL := fmt.Sprintf("http://cafe.example.com:%d%s", port, "/coffee")
+
+				Eventually(
+					func() error {
+						return expectRequestToSucceed(baseURL, address, "URI: /coffee")
+					}).
+					WithTimeout(timeoutConfig.RequestTimeout).
+					WithPolling(500 * time.Millisecond).
+					Should(Succeed())
+			})
+		})
+
+		Context("nginx directives", func() {
+			var conf *framework.Payload
+			snippetsFilterFilePrefix := "/etc/nginx/includes/SnippetsFilter_"
+
+			mainContext := fmt.Sprintf("%ssmain_", snippetsFilterFilePrefix)
+			httpContext := fmt.Sprintf("%shttp_", snippetsFilterFilePrefix)
+			httpServerContext := fmt.Sprintf("%shttp.server_", snippetsFilterFilePrefix)
+			httpServerLocationContext := fmt.Sprintf("%shttp.server.location_", snippetsFilterFilePrefix)
+
+			httpRouteSuffix := fmt.Sprintf("%s_all-contexts.conf", namespace)
+			grpcRouteSuffix := fmt.Sprintf("%s_grpc-all-contexts.conf", namespace)
+
+			BeforeAll(func() {
+				podNames, err := framework.GetReadyNGFPodNames(k8sClient, ngfNamespace, releaseName, timeoutConfig.GetTimeout)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(podNames).To(HaveLen(1))
+
+				ngfPodName := podNames[0]
+
+				conf, err = resourceManager.GetNginxConfig(ngfPodName, ngfNamespace)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			DescribeTable("are set properly for",
+				func(expCfgs []framework.ExpectedNginxField) {
+					for _, expCfg := range expCfgs {
+						Expect(framework.ValidateNginxFieldExists(conf, expCfg)).To(Succeed())
+					}
+				},
+				Entry("HTTPRoute", []framework.ExpectedNginxField{
+					{
+						Directive: "worker_priority",
+						Value:     "0",
+						File:      fmt.Sprintf("%s%s", mainContext, httpRouteSuffix),
+					},
+					{
+						Directive: "aio",
+						Value:     "on",
+						File:      fmt.Sprintf("%s%s", httpContext, httpRouteSuffix),
+					},
+					{
+						Directive: "auth_delay",
+						Value:     "10s",
+						File:      fmt.Sprintf("%s%s", httpServerContext, httpRouteSuffix),
+					},
+					{
+						Directive: "include",
+						Value:     fmt.Sprintf("%s%s", httpServerContext, httpRouteSuffix),
+						Servers:   []string{"cafe.example.com"},
+						File:      "http.conf",
```

Comment:
Is the Value and File swapped?

Reply:
No, 

Basically snippetsFilter are written in separate files. We use the file name with the directive `include` to add it to the configuration files

So we have a file `SnippetsFilter_main_default_all-context.conf` with contents

```
worker_priority 0;
```

Then in the nginx.conf, we use (in the context defined)

```
include SnippetsFilter_main_default_all-context.conf
```

These are missing include directive checks for `main` and `http` directives. Adding them currently